packages/client/src/components/OrderDataCompletion/OrderDataCompletion.tsx

```diff
@@ -108,11 +112,81 @@ const OrderDataCompletion = () => {
     setFloor(addressObject.floor || '');
     setPostcode(addressObject.postcode);
     setCity(addressObject.city);
+    setOldStateAddress([
+      addressObject.street,
+      addressObject.streetNumber,
+      addressObject.apartmentNumber || '',
+      addressObject.floor || '',
+      addressObject.postcode,
+      addressObject.city,
+    ]);
+  };
+
+  const handleSelectDeliveryHours = (e: SelectChangeEvent<string>) => {
+    e.preventDefault();
+    setDeliveryHours(e.target.value);
+    setDeliveryHourStart(e.target.value);
+  };
+
+  const userssProfileApi = new UserssProfileApi(apiConfiguration);
+  const updateUserProfile = async (updateData: UpdateUserDto) => {
+    try {
+      await userssProfileApi.update({ updateUserDto: updateData });
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
+  };
+
+  const usersAddressesApi = new UsersAddressesApi(apiConfiguration);
+  const createUserAddress = async (updateData: CreateAddressDto) => {
+    try {
+      const getAddressApi = await usersAddressesApi.create({ createAddressDto: updateData });
+      setAddressId(getAddressApi.id);
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
   };
 
+  // check if user put new address,
+  useEffect(() => {
+    const newAddressTable = [street, streetNumber, apartmentNumber, floor, postcode, city];
+    setAddress({ street, postcode, streetNumber, apartmentNumber, city });
+    // checking if address is the same like in checked radio button, control empty click on textfields.
+    if (JSON.stringify(oldStateAddress) === JSON.stringify(newAddressTable)) setStateChangedAddress(false);
+    else setStateChangedAddress(true);
```

Comment:
```TS
if (condition) func(false)
else func(true)
```

Można uprościć do:
```TS
func(!condition)
```

Tzn.
```TS
setStateChangedAddress(JSON.stringify(oldStateAddress) !== JSON.stringify(newAddressTable))
```

Oprócz tego, jeśli obiekt jest dość prosty (np. tutaj `string[]`) to możesz zamiast JSONa skorzystać z [`isEqual`](https://lodash.com/docs/4.17.15#isEqual).